presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusRecordCursor.java

```diff
@@ -0,0 +1,321 @@
+            case 1:
+                return fields.timestamp;
+            case 2:
+                return fields.value;
+        }
+        return null;
+    }
+
+    @Override
+    public boolean getBoolean(int field)
+    {
+        return true;
+    }
+
+    @Override
+    public long getLong(int field)
+    {
+        Type type = getType(field);
+        if (type.equals(TIMESTAMP)) {
+            return ((Timestamp) getFieldValue(field)).toInstant().toEpochMilli();
+        }
+        else {
+            throw new PrestoException(NOT_SUPPORTED, "Unsupported type " + getType(field));
+        }
+    }
+
+    @Override
+    public double getDouble(int field)
+    {
+        checkFieldType(field, DOUBLE);
+        return (double) getFieldValue(field);
+    }
+
+    @Override
+    public Slice getSlice(int field)
+    {
+        checkFieldType(field, createUnboundedVarcharType());
+        return Slices.utf8Slice((String) getFieldValue(field));
+    }
+
+    @Override
+    public Object getObject(int field)
+    {
+        return getFieldValue(field);
+    }
+
+    @Override
+    public boolean isNull(int field)
+    {
+        checkArgument(field < columnHandles.size(), "Invalid field index");
+        if (getFieldValue(field) == null) {
+            return true;
+        }
+        return false;
+    }
+
+    private void checkFieldType(int field, Type expected)
+    {
+        Type actual = getType(field);
+        checkArgument(actual.equals(expected), "Expected field %s to be type %s but is %s", field, expected, actual);
+    }
+
+    /**
+     * @param results a list of results from Prometheus, deserialized from JSON response, time series values are in inside list
+     * @return unwrapped inside time series list with each item in returned list having metric labels repeated as item labels
+     */
+    private List<PrometheusStandardizedRow> prometheusResultsInStandardizedForm(List<PrometheusMetricResult> results)
+    {
+        return results.stream().map(result ->
+                result.getTimeSeriesValues().getValues().stream().map(prometheusTimeSeriesValue ->
+                        new PrometheusStandardizedRow(
+                                getBlockFromMap(columnHandles.get(0).getColumnType(), metricHeaderToMap(result.getMetricHeader())),
+                                prometheusTimeSeriesValue.getTimestamp(),
+                                Double.parseDouble(prometheusTimeSeriesValue.getValue())))
+                        .collect(Collectors.toList()))
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+    }
+
+    private Map<String, String> metricHeaderToMap(Map<String, String> mapToConvert)
+    {
+        return ImmutableMap.<String, String>builder().putAll(mapToConvert).build();
+    }
+
+    /**
+     * Encodes the given map into a Block.
+     *
+     * @param mapType Presto type of the map
+     * @param map     Map of key/value pairs to encode
+     * @return Presto Block
```

Comment:
I would remove all javadocs content like:
```
     * @param mapType Presto type of the map
     * @param map     Map of key/value pairs to encode
     * @return Presto Block
```
It does not add much value.

Reply:
Done in fbb42d3e